tools/similarity_search/test/index.spec.ts

```diff
@@ -1,22 +1,98 @@
-import { SELF } from "cloudflare:test"
+import { SELF, env } from "cloudflare:test"
 import { describe, it, expect } from "vitest"
 
 import "../src/index"
 
+interface SimilarityCheckResponse {
+  similarity_score: number
+}
+
 describe("Authentication", () => {
-  it("returns 401 Unauthorized when API key is missing or invalid", async () => {
+  it("returns 401 Unauthorized when API key is invalid", async () => {
     const response = await SELF.fetch("https://example.com/", {
       method: "POST",
       headers: {
-        "Content-Type": "application/json"
+        "Content-Type": "application/json",
+        "X-API-Key": "invalid-api-key",
       },
       body: JSON.stringify({
         text: "Sample text",
-        namespace: "test-namespace"
-      })
+        namespace: "test-namespace",
+      }),
     })
 
     expect(response.status).toBe(401)
     expect(await response.text()).toBe("Unauthorized")
```

Comment:
I wouldn't assert specific wording for errors. A worker could be using a standard library that says something different for a 401 error.

Reply:
Done